src/Adapters/JobsAdapter.ts

```diff
@@ -0,0 +1,156 @@
+import ADTJobsData from '../Models/Classes/AdapterDataClasses/ADTJobsData';
+import AdapterMethodSandbox from '../Models/Classes/AdapterMethodSandbox';
+import AdapterResult from '../Models/Classes/AdapterResult';
+import {
+    AdapterMethodParamForCreateJobs,
+    AdapterMethodParamsForJobs,
+    ADT_ApiVersion,
+    AxiosObjParam,
+    IAuthService,
+    IJobsAdapter,
+    LOCAL_STORAGE_KEYS
+} from '../Models/Constants';
+import { applyMixins, validateExplorerOrigin } from '../Models/Services/Utils';
+import ADT3DSceneAdapter from './ADT3DSceneAdapter';
+import ADTAdapter from './ADTAdapter';
+
+const forceCORS = localStorage.getItem(
+    LOCAL_STORAGE_KEYS.FeatureFlags.Proxy.forceCORS
+);
+
+export default class JobsAdapter implements IJobsAdapter {
+    public adtHostUrl: string;
+    public authService: IAuthService;
+    public uniqueObjectId: string;
+    public tenantId: string;
+    protected useProxy: boolean;
+    protected adtProxyServerPath: string;
+
+    constructor(
+        adtHostUrl: string,
+        authService: IAuthService,
+        adtProxyServerPath = '/proxy/adt',
+        useProxy = true
+    ) {
+        this.setAdtHostUrl(adtHostUrl); // this should be the host name of the instance
+        this.adtProxyServerPath = adtProxyServerPath;
+        this.authService = authService;
+        /**
+         * Check if class has been initialized with CORS enabled or if origin matches dev or prod explorer urls,
+         * override if CORS is forced by feature flag
+         *  */
+        this.useProxy =
+            (useProxy || !validateExplorerOrigin(window.origin)) && !forceCORS;
+    }
+    setAdtHostUrl(hostName: string) {
+        if (hostName.startsWith('https://')) {
+            hostName = hostName.replace('https://', '');
+        }
+        this.adtHostUrl = hostName;
+    }
+    generateUrl(path: string) {
+        if (this.useProxy) {
+            return `${this.adtProxyServerPath}${path}`;
+        } else {
+            return `https://${this.adtHostUrl}${path}`;
+        }
+    }
+    generateHeaders(headers: AxiosObjParam = {}) {
+        if (this.useProxy) {
+            return {
+                ...headers,
+                'x-adt-host': this.adtHostUrl
+            };
+        } else {
+            return headers;
+        }
+    }
+    getAdtHostUrl() {
```

Comment:
I think you should ADTAdapter to this mixin array so that you will not need to implement ADTAdapter methods like getAdtHostUrl or setAdtHostUrl
https://www.typescriptlang.org/docs/handbook/mixins.html#alternative-pattern